pip_audit/_state.py

```diff
@@ -99,24 +97,20 @@ def finalize(self) -> None:
         raise NotImplementedError  # pragma: no cover
 
 
-class AuditSpinner(_StateActor, BaseSpinner):  # pragma: no cover
+class AuditSpinner(_StateActor):  # pragma: no cover
     """
-    A progress spinner for the `pip-audit` CLI, specialized from `BaseSpinner`.
-
-    This spinner is also written as a `AuditState` actor.
+    A progress spinner for `pip-audit`, using `rich.status`'s spinner support
+    under the hood.
     """
 
-    def __init__(self, message: str = "", **kwargs: Dict[str, Any]):
+    def __init__(self, message: str = "") -> None:
         """
-        Create a new `AuditSpinner`.
-
-        `message` is the initial text that the progress spinner should display.
-
-        Any remaining keyword arguments are forwarded onto the constructor of the underlying
-        `BaseSpinner` implementation.
+        Initialize the `AuditSpinner`.
         """
 
-        super().__init__(message=message, **kwargs)
+        self._console = Console()
+        # NOTE: audits can be quite fast, so we need a pretty high refresh rate here.
+        self._spinner = self._console.status(message, spinner="line", refresh_per_second=30)
```

Comment:
N.B.: If we wanted to be extra fancy, we could allow users to configure the `spinner=...` here, rather than using the same "line" spinner that `progress` uses.

Rich has a decent collection of spinners to choose from: 

https://github.com/Textualize/rich/blob/aa7926c1431eebfb2ccaab9f3b63a4ac6cd8dfe6/rich/_spinners.py#L22